Sources/Components/Dialogue/DialogueView.swift

```diff
@@ -147,25 +138,42 @@ public class DialogueView: UIView {
             detail.leadingAnchor.constraint(equalTo: leadingAnchor, constant: .largeSpacing),
             detail.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -.largeSpacing),
 
-            link.topAnchor.constraint(equalTo: detail.bottomAnchor, constant: .mediumSpacing),
+            primaryButton.topAnchor.constraint(equalTo: detail.bottomAnchor, constant: .mediumLargeSpacing),
+            primaryButton.centerXAnchor.constraint(equalTo: centerXAnchor),
+
+            link.topAnchor.constraint(equalTo: primaryButton.bottomAnchor, constant: .mediumSpacing),
             link.leadingAnchor.constraint(equalTo: leadingAnchor, constant: .largeSpacing),
             link.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -.largeSpacing),
-
-            primaryButton.topAnchor.constraint(equalTo: link.bottomAnchor, constant: .mediumLargeSpacing),
-            primaryButton.centerXAnchor.constraint(equalTo: centerXAnchor),
             ])
     }
 
+    func heightWithConstrainedWidth(width: CGFloat) -> CGFloat {
+        guard let text = detail.text else {
+            return 0
+        }
+
+        let constraintRect = CGSize(width: width, height: .greatestFiniteMagnitude)
+        let boundingBox = text.boundingRect(with: constraintRect, options: [.usesLineFragmentOrigin], attributes: [NSAttributedString.Key.font: detail.font], context: nil)
+
+        var moreSpacing: CGFloat = 0
+
+        if UIDevice.isSmallScreen() {
+            moreSpacing = .largeSpacing
+        }
+
+        if UIDevice.isLandscape() {
```

Comment:
Should this be an `else if UIDevice.isLandscape` instead? It looks like they are exclusive.